vpr/src/route/connection_router.cpp

```diff
@@ -8,7 +8,7 @@
 //as valid start locations for the current connection.
 //
 //Returns either the last element of the path, or nullptr if no path is found
-t_heap* ConnectionRouter::timing_driven_route_connection_from_route_tree(
+std::pair<bool, t_heap> ConnectionRouter::timing_driven_route_connection_from_route_tree(
```

Comment:
Can you clarify what's the motivation for changing the return value?

Reply:
It ensures that all memory allocated by the `ConnectionRouter` stays within the connection router.  The previous code transfers ownership to the caller, who would read some fields and then immediately free it.